src/schedule/edf_schedule.c

```diff
@@ -74,30 +70,18 @@ static void edf_scheduler_run(void *data)
 		    task->state != SOF_TASK_STATE_RUNNING)
 			continue;
 
-		edf_pdata = edf_sch_get_pdata(task);
+		deadline = task_get_deadline(task);
 
-		deadline = edf_pdata->deadline;
-
-		if (current >= deadline &&
-		    !(task->flags & SOF_SCHEDULE_FLAG_IDLE)) {
+		if (deadline == SOF_TASK_DEADLINE_NOW) {
 			/* task needs to be scheduled ASAP */
 			task_next = task;
 			break;
 		}
 
-		delta = deadline - current;
-
-		/* get highest priority */
-		if (task->priority < priority_next) {
-			priority_next = task->priority;
-			delta_next = delta;
+		/* get earliest deadline */
+		if (deadline < deadline_next || !task_next) {
```

Comment:
@tlauda is the check for !task_next really needed? If it were ever not NULL, we'd have quit the loop no?

Reply:
Done.